### PR TITLE
serials: adapt item JSON schema for UI

### PR DESCRIPTION
**Pull request by zannkukai**
* Adapts the item JSON schema to have a correct display into the UI
editor.
* Adapts serial preview API url according to REST guidelines.
* Adds constants for item issue status instead of strings.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Dependencies

Must be tested with https://github.com/rero/rero-ils-ui/pull/272

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/modules/holdings/api_views.py [hunk 1]
```diff
@@ -117,28 +117,32 @@ def pattern_preview():
...
-    holding = Holding.get_record_by_pid(data.get('holdings_pid'))
+    holding = Holding.get_record_by_pid(holding_pid)
     if not holding:
-        abort(404)
+        abort(404, "Holding not found")
     # librarian of same holdings library may receive issues
     # system librarians may receive for all libraries of organisation.
     if not can_receive_regular_issue(holding):
         abort(401)
+    # receive is allowed only on holdings of type serials and regular frequency
```
Comment (BadrAly) [inline rero_ils/modules/holdings/api_views.py:142]: remove this, it is already checked in the method `receive_regular_issue ` and
returns as well 400

File: rero_ils/modules/items/api/api.py [hunk 2]
```diff
@@ -104,6 +104,9 @@ def reasons_not_to_delete(self):
         links = self.get_links_to_me()
         if links:
             cannot_delete['links'] = links
+        if self.item_record_type == 'issue' \
```
Comment (BadrAly) [inline rero_ils/modules/items/api/api.py:107]: so you want to prevent the deletion of the `deleted_issues`?

Comment (zannkukai) [reply rero_ils/modules/items/api/api.py:107]: It's used to disable the "trash" delete button when an issue has already the deleted status

File: rero_ils/modules/holdings/api_views.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/holdings/api_views.py]: done

File: rero_ils/modules/items/api/api.py (hunk 2 above)
Comment (BadrAly) [reply rero_ils/modules/items/api/api.py:107]: by doing this, we will not be able to delete an `issue` with a `deleted` status. if it is not the case, ignore my message.

Comment (zannkukai) [reply rero_ils/modules/items/api/api.py:107]: I think we could *never* delete a received issue, even if the status is deleted.

Comment (BadrAly) [reply rero_ils/modules/items/api/api.py:107]: hmmm... a good question for the PO.

File: rero_ils/modules/items/api/issue.py [hunk 3]
```diff
@@ -42,3 +42,8 @@ def display_text(self):
     def issue_status(self):
         """Shortcut for issue status."""
         return self.get('issue', {}).get('status')
+
+    @property
+    def issue_regular(self):
+        """Shortcut for issue is regular."""
+        return self.get('issue', {}).get('regular', True)
```
Comment (lauren-d) [inline rero_ils/modules/items/api/issue.py:49]: cosmetic: rename boolean function like `is_regular_issue` ?

File: rero_ils/modules/items/permissions.py [hunk 4]
```diff
@@ -21,17 +21,37 @@
...
+            if record.item_record_type == 'issue' and record.issue_regular:
+                return False
+            if patron.organisation_pid == record.organisation_pid:
+                return check_patron_library_permissions(patron, record)
+        return False
+    return type('Check', (), {'can': can})()
+
+
+def can_update_item_factory(record, *args, **kwargs):
+    """Check if logged uuser can update an item.
```
Comment (lauren-d) [inline rero_ils/modules/items/permissions.py:45]: Typo: `uuser` to `user`

File: tests/utils.py [hunk 5]
```diff
@@ -76,8 +76,10 @@ def postdata(client, endpoint, data=None, headers=None):
             ('Accept', 'application/json'),
             ('Content-Type', 'application/json')
         ]
+    if url_data is None:
+        url_data = {}
```
Comment (blankoworld) [inline tests/utils.py:80]: Why not adding directly `url_data={}` in postdata method declaration?

File: tests/api/test_items_issue.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2020 RERO
```
Comment (iGormilhit) [inline tests/api/test_items_issue.py:4]:
```suggestion
# Copyright (C) 2020 RERO
# Copyright (C) 2020 UCLouvain
```
And you can do it in every file that you're updating.

**Pull request merged**